src/SignalR/clients/ts/signalr/src/HubConnectionBuilder.ts

```diff
@@ -13,6 +13,32 @@ import { JsonHubProtocol } from "./JsonHubProtocol";
 import { NullLogger } from "./Loggers";
 import { Arg, ConsoleLogger } from "./Utils";
 
+// This is exported from this module but NOT from index, so it's "internal" for testing purposes
+export const LogLevelNameMapping = {
+    // TS Lint prefers alphabetical sorting...
+    critical: LogLevel.Critical,
+    debug: LogLevel.Debug,
+    error: LogLevel.Error,
+    info: LogLevel.Information,
+    information: LogLevel.Information,
+    none: LogLevel.None,
+    trace: LogLevel.Trace,
+    warn: LogLevel.Warning,
+    warning: LogLevel.Warning,
```

Comment:
Should we capitalize the first letter like we do for the LogLevel enum?

Reply:
If not, we might want to change our HubConnection state enum to use all lower-case characters too:

https://github.com/aspnet/AspNetCore/blob/4964b338abe02bdbb21ff69928d38b11e85c90c9/src/SignalR/clients/ts/signalr/src/HubConnection.ts#L17-L28

Reply:
It needs to be lowercase for the parsing to work since we do `name.toLowerCase()`. This type, and it's members, don't appear to users, it's just an internal type used for the parser and for the tests.

Reply:
OK. I see it's only used as input to configureLogging.

So in JavaScript, you could capitalize the string however you want, but in TypeScript you're still limited to lowercase strings because configureLogging expects a `keyof typeof LogLevelNameMapping`.